bot/resources/tags/customcooldown.md

```diff
@@ -0,0 +1,22 @@
+**Cooldowns in discord.py**
+
+Cooldowns can be used in discord.py to rate-limit. In this example, we're using it in an on_message.
+
+```python
+from discord.ext import commands
+
+message_cooldown = commands.CooldownMapping.from_cooldown(1.0, 60.0, commands.BucketType.user)
+
+@bot.event
+async def on_message(message):
+    bucket = message_cooldown.get_bucket(message)
+    # update_rate_limit returns a time you need to wait before
+    # trying again
+    retry_after = bucket.update_rate_limit()
+    if retry_after:
+        await message.channel.send("Slow down! You're sending messages too fast")
```

Comment:
Using `retry_after ` in the message demonstrates what the variable means in a clean, succinct way compared to a comment. The f-string may confuse some, but I think it's a worthy trade-off.
```suggestion
    retry_after = bucket.update_rate_limit()
    if retry_after:
        await message.channel.send(f"Slow down! Try again in {retry_after} seconds.")
```